sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcFileFormat.scala

```diff
@@ -153,10 +152,9 @@ class OrcFileFormat
       } else {
         // ORC predicate pushdown
         if (orcFilterPushDown && filters.nonEmpty) {
-          OrcUtils.readCatalystSchema(filePath, conf, ignoreCorruptFiles).foreach { fileSchema =>
```

Comment:
It seems `OrcUtils.readCatalystSchema` catch the `FileFormatException`.
I think this is the key the code uses `OrcUtils.readCatalystSchema` not `orcSchema`.

Reply:
thanks for your review @beliefer , you are right, `OrcUtils.readCatalystSchema` catch the `FileFormatException`. But if `OrcUtils.readCatalystSchema` catches an `FileFormatException`, it should have been thrown on line 146.

https://github.com/apache/spark/blob/18ca369f01905b421a658144e23b5a4e60702655/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcFileFormat.scala#L146-L161


https://github.com/apache/spark/blob/18ca369f01905b421a658144e23b5a4e60702655/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcUtils.scala#L72-L94

Reply:
In what scenario does line 147 in `OrcFileFormat` (`Utils.tryWithResource(OrcFile.createReader(filePath, readerOptions))(_.getSchema) `) not throw an `FileFormatException`,  but line 77 in `OrcUtils`(`Utils.tryWithResource(OrcFile.createReader(file, readerOptions)) { reader => reader.getSchema } `) does?

Reply:
And should we change to use `OrcUtils.readSchema(filePath, conf, ignoreCorruptFiles)` at line 147 in `OrcFileFormat`, then it seems that the `FileFormatException ` is handled in `OrcFileFormat`, otherwise, the `FileFormatException ` will be thrown and handled by `FileScanRDD.`  Am I right? 
`